src/PointwiseFunctions/AnalyticSolutions/GrMhd/SmoothFlow.hpp

```diff
@@ -106,11 +106,64 @@ class SmoothFlow {
                  hydro::Tags::Pressure<DataType>,
                  hydro::Tags::MagneticField<DataType, 3, Frame::Inertial>>;
 
-  /// Retrieve the primitive variables at time `t` and spatial coordinates `x`
+  // Primitive variables.
   template <typename DataType>
-  tuples::tagged_tuple_from_typelist<variables_tags<DataType>> variables(
+  tuples::TaggedTuple<hydro::Tags::RestMassDensity<DataType>> variables(
       const tnsr::I<DataType, 3>& x, double t,
-      variables_tags<DataType> /*meta*/) const noexcept;
+      tmpl::list<hydro::Tags::RestMassDensity<DataType>> /*meta*/) const
+      noexcept;
+
+  template <typename DataType>
+  tuples::TaggedTuple<hydro::Tags::SpecificInternalEnergy<DataType>> variables(
+      const tnsr::I<DataType, 3>& x, double t,
+      tmpl::list<hydro::Tags::SpecificInternalEnergy<DataType>> /*meta*/) const
+      noexcept;
+
+  template <typename DataType>
+  tuples::TaggedTuple<hydro::Tags::Pressure<DataType>> variables(
+      const tnsr::I<DataType, 3>& x, double /*t*/,
+      tmpl::list<hydro::Tags::Pressure<DataType>> /*meta*/) const noexcept;
+
+  template <typename DataType>
+  tuples::TaggedTuple<
+      hydro::Tags::SpatialVelocity<DataType, 3, Frame::Inertial>>
+  variables(const tnsr::I<DataType, 3>& x, double /*t*/,
+            tmpl::list<hydro::Tags::SpatialVelocity<
+                DataType, 3, Frame::Inertial>> /*meta*/) const noexcept;
+
+  template <typename DataType>
+  tuples::TaggedTuple<hydro::Tags::MagneticField<DataType, 3, Frame::Inertial>>
+  variables(const tnsr::I<DataType, 3>& x, double /*t*/,
+            tmpl::list<hydro::Tags::MagneticField<
+                DataType, 3, Frame::Inertial>> /*meta*/) const noexcept;
+
+  template <typename DataType>
+  tuples::TaggedTuple<hydro::Tags::DivergenceCleaningField<DataType>> variables(
+      const tnsr::I<DataType, 3>& x, double /*t*/,
+      tmpl::list<hydro::Tags::DivergenceCleaningField<DataType>> /*meta*/) const
+      noexcept;
+
+  template <typename DataType>
+  tuples::TaggedTuple<hydro::Tags::LorentzFactor<DataType>> variables(
+      const tnsr::I<DataType, 3>& x, double /*t*/,
+      tmpl::list<hydro::Tags::LorentzFactor<DataType>> /*meta*/) const noexcept;
+
+  template <typename DataType>
+  tuples::TaggedTuple<hydro::Tags::SpecificEnthalpy<DataType>> variables(
+      const tnsr::I<DataType, 3>& x, double t,
+      tmpl::list<hydro::Tags::SpecificEnthalpy<DataType>> /*meta*/) const
+      noexcept;
+
+  template <typename DataType, typename... Tags>
```

Comment:
document all the variables functions for each solution grouping the ones for each solution